include/piece/king.h

```diff
@@ -5,17 +5,13 @@
 #include <set>
 
 class King : public Piece {
-    std::vector<std::pair<int, int>> directions
-        = { { 1, 0 }, { -1, 0 }, { 0, 1 }, { 0, -1 }, { 1, 1 },
-              { 1, -1 }, { -1, 1 }, { -1, -1 } };
 
 public:
     King(Color color, Coordinate location, PieceType type,
         std::shared_ptr<Board> board);
-    bool is_valid_move(Coordinate square) override;
-    bool is_check(std::set<Coordinate>& attacked_squares);
+    void get_threatened_squares(std::set<Coordinate>& s) override;
```

Comment:
do you need to do override get valid moves

Reply:
resolved.